clients/python-wrapper/lakefs/config.py

```diff
@@ -60,35 +107,162 @@ def __init__(self, verify_ssl: Optional[bool] = None, proxy: Optional[str] = Non
 
         if kwargs:
             return
+        self._load_from_config_file()
+        self._load_from_environment()
+        # Check for IAM provider if no session token and no credentials
+        if self.access_token is None and self.username is None and self.password is None:
+            self._iam_provider = get_iam_provider_from_env_or_file(self._config_data)
 
```

Comment:
I'll tell you why I'm confused:
On the one hand you create a load from config and load from environment and split the code in a way where you load the configuration based on the source.
But then you create functions that get dedicated information get iam provider / get aws iam profile which provide specific configuration information but go over all sources.
For clarity - I think we should choose one approach for collecting configuration information.
I have to admit I got dizzy trying to read this